group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroupMember.java

```diff
@@ -0,0 +1,756 @@
+     */
+    public Map<String, Set<Integer>> assignedWarmupTasks() {
+        return assignedWarmupTasks;
+    }
+
+    /**
+     * @return The set of active tasks awaiting revocation from the member.
+     */
+    public Map<String, Set<Integer>> activeTasksPendingRevocation() {
+        return activeTasksPendingRevocation;
+    }
+
+    /**
+     * @return The set of standby tasks awaiting revocation from the member.
+     */
+    public Map<String, Set<Integer>> standbyTasksPendingRevocation() {
+        return standbyTasksPendingRevocation;
+    }
+
+    /**
+     * @return The set of warmup tasks awaiting revocation from the member.
+     */
+    public Map<String, Set<Integer>> warmupTasksPendingRevocation() {
+        return warmupTasksPendingRevocation;
+    }
+
+    /**
+     * @param targetAssignment The target assignment of this member in the corresponding group.
+     *
+     * @return The StreamsGroupMember mapped as StreamsGroupDescribeResponseData.Member.
+     */
+    public StreamsGroupDescribeResponseData.Member asStreamsGroupDescribeMember(
+        Assignment targetAssignment
+    ) {
+        final StreamsGroupDescribeResponseData.Assignment describedTargetAssignment =
+            new StreamsGroupDescribeResponseData.Assignment();
+
+        if (targetAssignment != null) {
+            describedTargetAssignment
+                .setActiveTasks(taskIdsFromMap(targetAssignment.activeTasks()))
+                .setStandbyTasks(taskIdsFromMap(targetAssignment.standbyTasks()))
+                .setWarmupTasks(taskIdsFromMap(targetAssignment.warmupTasks()));
+        }
+
+        return new StreamsGroupDescribeResponseData.Member()
+            .setMemberEpoch(memberEpoch)
+            .setMemberId(memberId)
+            .setAssignment(
+                new StreamsGroupDescribeResponseData.Assignment()
+                    .setActiveTasks(taskIdsFromMap(assignedActiveTasks))
+                    .setStandbyTasks(taskIdsFromMap(assignedStandbyTasks))
+                    .setWarmupTasks(taskIdsFromMap(assignedWarmupTasks)))
+            .setTargetAssignment(describedTargetAssignment)
+            .setClientHost(clientHost)
+            .setClientId(clientId)
+            .setInstanceId(instanceId)
+            .setRackId(rackId)
+            .setClientTags(clientTags.entrySet().stream().map(
+                entry -> new StreamsGroupDescribeResponseData.KeyValue()
+                    .setKey(entry.getKey())
+                    .setValue(entry.getValue())
+            ).collect(Collectors.toList()))
+            .setProcessId(processId)
+            .setTopologyEpoch(topologyEpoch)
+            .setUserEndpoint(
+                userEndpoint == null ? null :
+                    new StreamsGroupDescribeResponseData.Endpoint()
+                        .setHost(userEndpoint.host())
+                        .setPort(userEndpoint.port())
+            );
+        // TODO: TaskOffset, TaskEndOffset, IsClassic are to be implemented.
```

Comment:
I think when merging to trunk, we should either implement the missing TODOs or create a ticket instead and remove the TODO.